test/src/test/java/hudson/tasks/ArtifactArchiverTest.java

```diff
@@ -248,6 +265,125 @@ public void testDefaultExcludesOff() throws Exception {
         assertTrue(artifacts.child("dir").child(".svn").child("file").exists());
     }
 
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathOK() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        archiver.setBasePath("module/dist");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals(Result.SUCCESS, build(p));
+        VirtualFile artifacts = p.getBuildByNumber(1).getArtifactManager().root();
+        assertTrue(artifacts.child("target").child("file").exists());
```

Comment:
please don't use raw junit assertions without a message.
Also consider if this fails is there enough information in the unit test output to see what is wrong?